app/src/main/java/org/mozilla/focus/fragment/WebFragment.java

```diff
@@ -107,11 +111,25 @@ public void onPause() {
             webView.saveWebViewState(session);
         }
 
+        if (AppConstants.isGeckoBuild() && dayPassedSinceLastUpload()) {
+            // Record new edited date
+            PreferenceManager.getDefaultSharedPreferences(getContext()).edit()
+                    .putLong(lastMobileMetricsPing, System.currentTimeMillis()).apply();
+            webView.getTelemetrySnapshots();
+        }
+
         webView.onPause();
 
         super.onPause();
     }
 
+
+    private boolean dayPassedSinceLastUpload() {
+        final long lastPing = PreferenceManager.getDefaultSharedPreferences(getContext()).getLong(lastMobileMetricsPing, 0);
+        // Make sure a minimum of 1 day has passed since we collected data
+        return System.currentTimeMillis() >= (lastPing + dayInMilliseconds);
+    }
```

Comment:
On desktop we check for "midnight locally has passed" - or in other words: If it's at least the next day for this user. You could just save YYYYMMDD and check if the value for today is larger.